arroyo-api/src/connection_tables.rs

```diff
@@ -133,81 +335,118 @@ pub(crate) async fn create(
         .await
         .map_err(|err| handle_db_error("connection_table", err))?;
 
-    transaction.commit().await.map_err(log_and_map)?;
-    Ok(())
-}
+    transaction.commit().await.map_err(log_and_map_rest)?;
 
-pub(crate) async fn delete(
-    req: DeleteConnectionTableReq,
-    auth: AuthData,
-    client: &impl GenericClient,
-) -> Result<(), Status> {
-    let deleted = api_queries::delete_connection_table()
-        .bind(client, &auth.organization_id, &req.id)
+    let table = api_queries::get_connection_table()
+        .bind(&client, &auth_data.organization_id, &pub_id)
+        .one()
         .await
-        .map_err(|e| handle_delete("connection_table", "pipelines", e))?;
-
-    if deleted == 0 {
-        return Err(Status::not_found(format!(
-            "No connection table with id {}",
-            req.id
-        )));
-    }
+        .map_err(log_and_map_rest)?
+        .try_into()
+        .map_err(log_and_map_rest)?;
 
-    Ok(())
+    Ok(Json(table))
 }
 
-pub(crate) async fn test(
-    req: CreateConnectionTableReq,
-    auth: AuthData,
-    client: &impl GenericClient,
-) -> Result<Receiver<Result<TestSourceMessage, Status>>, Status> {
-    let (connector, _, config, schema) = get_and_validate_connector(&req, &auth, client).await?;
+impl TryInto<ConnectionTable> for DbConnectionTable {
+    type Error = ();
+    fn try_into(self) -> Result<ConnectionTable, Self::Error> {
+        let Some(connector) = connector_for_type(&self.connector) else {
+                warn!("invalid connector {} in saved ConnectionTable {}", self.connector, self.id);
+                return Err(());
+            };
 
-    let (tx, rx) = channel(8);
+        let connection = get_connection_profile(&self, &*connector);
+
+        let table = serde_json::to_string(&self.config).unwrap();
+        let schema: Option<ConnectionSchema> = self.schema.and_then(|s| {
+            serde_json::from_value(s)
+                .map_err(|e| {
+                    warn!("Error during connection schema deserialization: {:?}", e);
+                })
+                .ok()
```

Comment:
I don't think this is necessarily the correct behavior — if we fail to deserialize the schema that will cause problems later and we should error out here instead of telling our caller that there was no schema

Reply:
True, I've made it return an error.